### PR TITLE
ID-9 – Set minimum of 10 characters for new raw passwords

### PR DESCRIPTION
**Pull request by NoelLH**

### CODE REVIEW

File: src/Domain/Person.php [hunk 1]
```diff
@@ -170,8 +172,15 @@ public function validateCaptchaAndRawPasswordSetIfNew(ExecutionContextInterface
         }
 
         // Entity brand new or somehow otherwise without a password, and none set.
-        if (empty($this->password) && empty($this->raw_password)) {
-            $context->buildViolation('Password is required to create an account')
+        $passwordMissing = (
```
Comment (alihejazi) [inline src/Domain/Person.php:175]: @NoelLH I'd probably rename this to something a bit more accurate like `$passwordMissingOrBlank` or `$passwordInvalid` or `$passwordMissingOrDoesNotMeetRequirements`.

File: src/Domain/Person.php [hunk 2]
```diff
@@ -170,8 +172,15 @@ public function validateCaptchaAndRawPasswordSetIfNew(ExecutionContextInterface
...
+            empty($this->password) &&
+            (empty($this->raw_password) || mb_strlen($this->raw_password) < static::MIN_PASSWORD_LENGTH)
```
Comment (alihejazi) [inline src/Domain/Person.php:177]: @NoelLH I think we should add some extra `()` to explicitly highlight if the `||` operator applies to `empty($this->password) && (empty($this->raw_password)` as a whole or just `(empty($this->raw_password)`.

File: tests/Application/Actions/CreatePersonTest.php [hunk 3]
```diff
@@ -147,8 +147,44 @@ public function testMissingData(): void
...
+        $payloadJSON = (string) $response->getBody();
+
+        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertJson($payloadJSON);
+
+        $expectedJSON = json_encode([
+            'error' => [
+                'description' => 'Validation error: Password of 10 or more characters is required to create an account',
                 'type' => 'BAD_REQUEST',
             ],
```
Comment (alihejazi) [inline tests/Application/Actions/CreatePersonTest.php:189]: @NoelLH should we also add a test to cover the edge case of white spaces in passwords? e.g. 10 spaces?

File: src/Domain/Person.php (hunk 2 above)
Comment (NoelLH) [reply src/Domain/Person.php:177]: @alihejazi I don't understand where these would go. To me it is already very clear and extraneous brackets would make it harder to read. There are already no dependencies on operator precedence, the brackets around the second condition line make it explicit.

File: tests/Application/Actions/CreatePersonTest.php (hunk 3 above)
Comment (NoelLH) [reply tests/Application/Actions/CreatePersonTest.php:189]: @alihejazi In my opinion, no. Spaces are valid characters and we shouldn't build in new constraints we didn't specify.
